src/Faker/Container/ContainerBuilder.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Faker\Container;
+
+use Faker\English\FileExtension;
+use Faker\Extension\File;
+use Psr\Container\ContainerInterface;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ * @experimental This class is experimental and does not fall under our BC promise
+ */
+class ContainerBuilder
+{
+    /**
+     * @var array
+     */
+    private $container = [];
+
+    /**
+     * @param string|callable|object $value
+     * @param string|null $name
+     */
+    public function add($value, string $name = null): self
+    {
+        if (!is_string($value) && !is_callable($value) && !is_object($value)) {
+            throw new \LogicException(sprintf('First argument to "%s::add()" must be a string, callable or object.', __CLASS__));
+        }
+
+        if ($name === null) {
+            if (is_string($value)) {
+                $name = $value;
+            } elseif (is_object($value)) {
+                $name = get_class($value);
+            } else {
+                throw new \LogicException(sprintf('Second argument to "%s::add()" is required not passing a string or object as first argument', __CLASS__));
+            }
+        }
+
+        $this->container[$name] = $value;
+
+        return $this;
+    }
+
+    public function build(): ContainerInterface
+    {
+        return new Container($this->container);
+    }
+
+    public static function getDefault(): ContainerInterface
```

Comment:
```suggestion
    public static function create(): ContainerInterface
```

Reply:
This could be confused with "build". 

I want to be clear that this builds the default container.. Maybe this should be part of `Generator` or `Factory` instead?

Reply:
I want the container to be immutable. Using a builder class is the simplest and easiest to understand. 

The alternative would be that the container builds itself and it is immutable only after we run `Container::build()`.